core/src/main/java/nl/nn/adapterframework/pipes/StringIteratorPipe.java

```diff
@@ -92,30 +93,30 @@ private boolean finalizeBlock() throws SenderException, TimeOutException, IOExce
 				if (processingInBlocks && isCombineBlocks() && itemCounter>0) {
 					itemCounter=0;
 					items.append(getBlockSuffix());
-					boolean result = super.handleItem(items.toString());
+					boolean result = super.handleItem(items.toString()) == null;
 					items.setLength(0);
 					return result;
 				}
 				return false;
 			}
 			
 			@Override
-			public boolean handleItem(String item) throws SenderException, TimeOutException, IOException {
+			public String handleItem(String item) throws SenderException, TimeOutException, IOException {
 				if (processInBlocksBySize && itemCounter==0) {
 					startBlock();
 				} 
 				if (processInBlocksByKey) {
 					String key = getKey(item);
 					if (!key.equals(previousKey)) { 
 						if (previousKey!=null && !finalizeBlock()) {
-							return false;
+							return PipeForward.SUCCESS_FORWARD_NAME;
```

Comment:
I think finalizeBlock should return the stopReason, and handleItem should return that here.
It would be good to create a test that hits this code.